src/Explorer/Controls/Settings/SettingsComponent.test.tsx

```diff
@@ -35,7 +46,10 @@ describe("SettingsComponent", () => {
       node: undefined,
       hashLocation: "settings",
       isActive: ko.observable(false),
-      onUpdateTabsButtons: undefined
+      onUpdateTabsButtons: undefined,
+      getPendingNotification: Q.Promise<DataModels.Notification>(() => {
```

Comment:
Remove new useage of Q promises wherever possible

Reply:
getPendingNotification generates a Q.Promise which is what is being used here. in settingsv1, the promise is resolved and only then was the tab opened. But in settingsv2, I am trying to push the call into the tab activation for a better experience. Hence this Q.Promise has to be propagated to the new code too